### ISSUE TITLE
Adding a `IamSlice` feature

### ISSUE THREAD

**Issue by danielhuppmann** [IAMconsortium/pyam#630]
# Description

We often want to know which elements exist on an dimension after filtering, e.g., which variables exist for a specific region (see https://github.com/IAMconsortium/nomenclature/pull/99). This can be done by the following

```python
vars = df.filter(region="Region A").variable
```

However, this is inefficient because this creates a full (downselected) copy of the (timeseries) `data` and `meta` tables.

# Proposed Solution

A new class **IamSlice** which is a derivative of the `pd.MultiIndex` of the internal `_data` pd.Series. The IamSlice is returned by the method `slice()`, which takes the same arguments as `filter()`.

## Expected usage

```python
vars = df.slice(region="Region A").variable
```

**Comment by gidden** [comment]
+1, great idea!

**Comment by danielhuppmann** [comment]
I like your idea about a boolean mask, but I would not how to implement it...

On the second idea about allowing `df[df.slice()]`, that sounds great and easily doable...

**Comment by danielhuppmann** [comment]
Closed via #637
